validator.js

```diff
@@ -202,6 +205,11 @@
         if (query && /\s/.test(query)) {
             return false;
         }
+
+        if (options.allow_protocol_relative_urls && protocolRelativeURLPrefix.test(url)) {
+            url = protocolRelativeURLPrefix.exec(url)[1];
```

Comment:
Why not `url = url.substr(2)`? then you can remove the `var protocolRelativeURLPrefix = /^\/\/(.*)/;` line above.